Makefile

```diff
@@ -109,7 +109,7 @@ check: vet
 
 .PHONY: test
 test:
-	go test -race -v -mod=mod -cover ${PACKAGES}
+	$(SUDO) -E go test -race -v -mod=mod -cover ${PACKAGES}
```

Comment:
Why do we need privilege when performing the unit test? At least for me, I don't have to go toolchain installed for `root` user 😂

Reply:
There are ProcessConfigurations tests in Unit test. And used `/var/lib/` as root path.

Reply:
Um. We'd better not pollute the developer's local file system. Directories base UT is better to be performed within a local temporary dir :-)

Reply:
So, I rewrote a SetupEnvironment function, the single test will not test this function. what do you think?

Reply:
SGTM